### PR TITLE
Random string custom charset

### PR DESCRIPTION
**Pull request by m3tomlins**
### reference
Today's Slack thread to enhance the existing randomString() function to support a conversion from JMeter's own ${__RandomString()} function. I have enhanced the documentation to describe the changes. Additionally, Tom mentioned that an inevitable version change to ks-jslib-utils to 1.2.0 would require additional edits to the documentation.

### technical explanation
Enhance the existing randomString() function to accept a second string parameter to provide customized charset. Refactor the internal charset to be wrapped in a conditional if statement, for backwards compatibility where a charset was not provided.
Provide a new randomString.js set of tests; leveraging suggestions from Tom and others. 

### other comments
This is my first PR as a contributor to any open source project. Please forgive any mistakes or missed steps in this process. Thank you.

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -13,10 +13,10 @@ export function randomItem(arrayOfItems){
   return arrayOfItems[Math.floor(Math.random() * arrayOfItems.length)];
 }
 
-export function randomString(length) {
-  const charset = 'abcdefghijklmnopqrstuvwxyz';
+export function randomString(length, charset='') {
+  if (!charset) charset = 'abcdefghijklmnopqrstuvwxyz';
```
Comment (mstoykov) [inline src/utils.js]: You can just set the default value directly in the function declaration.
```suggestion
export function randomString(length, charset='abcdefghijklmnopqrstuvwxyz') {
```

**Review by mstoykov — CHANGES_REQUESTED** [review]
Thanks for this PR @norillsa :bow: 

LGTM in general, and thank you for adding a bunch of tests :bow: 
I have a small comment on the implementation, after which I think we can merge it.

File: src/utils.js (hunk 1 above)
Comment (m3tomlins) [reply src/utils.js]: Sure - that checks out fine - I actually had pulled the example from Tom's Slack reply.

Also, much credit to him for many of the test cases also.

**Review by mstoykov — APPROVED** [review]
:bow:

**Comment by javaducky** [comment]
Thanks for your first contribution @norillsa ! 

All LGTM, but I'll let someone who really knows what's up with Javascript complete a review 😅...I just wanted to say thanks.

**Comment by tom-miseur** [comment]
I'd like to have @sniku comment/review before merging if that's OK. He should be back next week.

**Review by sniku — APPROVED** [review]
Thanks for the contribution. It looks to be backward compatible. :tada: 
Please sign the CLA, then we can merge it and make a new release.
After that, we can update the docs.

**Review by tom-miseur — APPROVED** [review]

**Comment by m3tomlins** [comment]
I just signed/agreed on the CLA - you should be good to go!

**Comment by javaducky** [comment]
@norillsa, looks like the issue is now that you probably changed your `user.name` config after your first 2 commits. So CLA is thinking there's 2 different authors. I think you'll need to rewrite history and do a `git commit --amend --reset-author` for those initial commits.

**Comment by m3tomlins** [comment]
@javaducky - I'm trying to rebase those 2 commits and get them over to my personal github "norillsa <mark@perfbytes>" as you suggested. Do I need to git push origin <branch> -f or something to get those to show up here?

And I think it was a --global author when I switched to another machine to work on this....that is fixed now with some separate folders:
    git.personal
    git.work

**Comment by m3tomlins** [comment]
Thanks @javaducky for the tips on rebase author on those commits. And thank you git log.

**Comment by javaducky** [comment]
CLA issues and history all appear to be resolved. LGTM, thanks @norillsa and congrats on your first PR! 🍾

**Pull request merged**